src/BlockTypes/ProductDetails.php

```diff
@@ -42,15 +42,41 @@ protected function render( $attributes, $content, $block ) {
 	 * @return string The tabs html to be rendered by the block
 	 */
 	protected function render_tabs() {
+		// This filter ensures the review form is properly rendered when the Product Details block is used alongside
+		// the Classic Single Product Block. It can be removed as soon as the Classic Single Product Block is deprecated.
+		add_filter( 'woocommerce_product_review_comment_form_args', array( $this, 'review_comment_form' ), 0 );
+
 		ob_start();
 
 		while ( have_posts() ) {
 			the_post();
 			woocommerce_output_product_data_tabs();
 		}
 
-		$tabs = ob_get_clean();
+		return ob_get_clean();
+	}
+
+	/**
+	 * Ensures the review form is properly rendered when the Product Details block is used alongside the Classic Single Product Block.
+	 *
+	 * This method can be removed as soon as the Classic Single Product Block is deprecated.
+	 *
+	 * @param array $comment_form The comment form arguments.
+	 *
+	 * @return array The updated comment form arguments.
+	 */
+	public function review_comment_form( $comment_form ) {
+		$comment_form['comment_field'] = '<div class="comment-form-rating"><label for="rating">' . esc_html__( 'Your rating', 'woo-gutenberg-products-block' ) . ( wc_review_ratings_required() ? '&nbsp;<span class="required">*</span>' : '' ) . '</label><select name="rating" id="rating-select" required>
```

Comment:
If we can fix the stars issue, I think that we should add [this logic](https://github.com/woocommerce/woocommerce/blob/3681b84a4c524691ad0748d29992a2d75df87fde/plugins/woocommerce/templates/single-product-reviews.php/#L123-L124) too.